framework/src/base/FEProblemBase.C

```diff
@@ -4038,7 +4038,8 @@ FEProblemBase::computeResidualType(const NumericVector<Number> & soln,
   }
   _aux->residualSetup();
 
-  _nl->computeTimeDerivatives();
+  if (type == Moose::KT_ALL || type == Moose::KT_TIME)
+    _nl->computeTimeDerivatives();
```

Comment:
There are equations, that couple time derivatives of other variables. These terms would appear in `KT_NONTIME` residual.  Are you sure that `coupledDot` will give the up-to-date time derivative with this change?

Reply:
I remembered why I add if statement this line because in CrankNicolson otherwise we will see dividing 0 dt. I guess if all time steppers give a valid initial dt like what I fixed in `FunctionDT`, we can change this back easily. We should think carefully here though.

Reply:
Yeah, looks like it will require more work if we want to remove unnecessary evaluations here. At least if a system does not require uDot, we should not add the vector in https://github.com/idaholab/moose/blob/devel/framework/src/base/NonlinearSystemBase.C#L117
and this evaluation can be avoid.